pkg/controllers/disruption/orchestration/queue.go

```diff
@@ -0,0 +1,307 @@
+func NewCommand(replacements []nodeclaim.Key, candidates []*state.StateNode, timeAdded time.Time, reason string, consolidationType string) *Command {
+	return &Command{
+		ReplacementKeys: lo.Map(replacements, func(key nodeclaim.Key, _ int) *NodeClaimKey {
+			return &NodeClaimKey{Key: key}
+		}),
+		Candidates:        candidates,
+		Method:            reason,
+		ConsolidationType: consolidationType,
+		TimeAdded:         timeAdded,
+	}
+}
+
+func (q *Queue) Name() string {
+	return "disruption.queue"
+}
+
+func (q *Queue) Builder(_ context.Context, m manager.Manager) controller.Builder {
+	return controller.NewSingletonManagedBy(m)
+}
+
+func (q *Queue) Reconcile(ctx context.Context, _ reconcile.Request) (reconcile.Result, error) {
+	// The queue depth is the number of commands currently being considered.
+	// This should not reference the RateLimitingInterface, as there can be items that
+	// haven't been requeued yet due to their per-item backoff.
+	disruptionQueueDepthGauge.Set(float64(q.depth))
+	// Check if the queue is empty. client-go recommends not using this function to gate the subsequent
+	// get call, but since we're popping items off the queue synchronously, there should be no synchonization
+	// issues.
+	for q.Len() == 0 {
+		time.Sleep(1 * time.Second)
+		// return reconcile.Result{RequeueAfter: 1 * time.Second}, nil
+	}
+	// Get command from queue. This waits until queue is non-empty.
+	item, shutdown := q.RateLimitingInterface.Get()
+	if shutdown {
+		panic("disruption queue has shut down")
+	}
+	cmd := item.(*Command)
+	defer q.RateLimitingInterface.Done(cmd)
+
+	if err := q.WaitOrTerminate(ctx, cmd); err != nil {
+		// If recoverable, re-queue and try again.
+		if !IsUnrecoverableError(err) {
+			// store the error that is causing us to fail so we can bubble it up later if this times out.
+			cmd.LastError = err
+			q.RateLimitingInterface.AddRateLimited(cmd)
+			return reconcile.Result{RequeueAfter: controller.Immediately}, nil
+		}
+		// If the command failed, bail on the action.
+		// 1. Emit metrics for launch failures
+		// 2. Ensure cluster state no longer thinks these nodes are deleting
+		// 3. Remove it from the Queue's internal data structure
+		failedLaunches := lo.Filter(cmd.ReplacementKeys, func(key *NodeClaimKey, _ int) bool {
+			return !key.Initialized
+		})
+		disruptionReplacementNodeClaimFailedCounter.With(map[string]string{
+			methodLabel:            cmd.Method,
+			consolidationTypeLabel: cmd.ConsolidationType,
+		}).Add(float64(len(failedLaunches)))
+
+		q.cluster.UnmarkForDeletion(lo.Map(cmd.Candidates, func(s *state.StateNode, _ int) string { return s.ProviderID() })...)
+		multiErr := multierr.Combine(err, state.RequireNoScheduleTaint(ctx, q.kubeClient, false, cmd.Candidates...), cmd.LastError)
+		q.Remove(cmd)
+		logging.FromContext(ctx).With("nodeNames", strings.Join(lo.Map(cmd.Candidates, func(s *state.StateNode, _ int) string {
```

Comment:
```suggestion
		logging.FromContext(ctx).With("nodes", strings.Join(lo.Map(cmd.Candidates, func(s *state.StateNode, _ int) string {
```

Our pattern has just to use the subject name and there's an implicit assumption that that value is a name